mono/metadata/marshal.c

```diff
@@ -264,12 +264,12 @@ mono_marshal_init (void)
 		register_icall (mono_gc_wbarrier_generic_nostore, "wb_generic", "void ptr", FALSE);
 		register_icall (mono_gchandle_get_target, "mono_gchandle_get_target", "object int32", TRUE);
 		register_icall (mono_marshal_isinst_with_cache, "mono_marshal_isinst_with_cache", "object object ptr ptr", FALSE);
-		register_icall (mono_threads_enter_gc_safe_region_unbalanced, "mono_threads_enter_gc_safe_region_unbalanced", "ptr ptr", TRUE);
-		register_icall (mono_threads_exit_gc_safe_region_unbalanced, "mono_threads_exit_gc_safe_region_unbalanced", "void ptr ptr", TRUE);
-		register_icall (mono_threads_enter_gc_unsafe_region_unbalanced, "mono_threads_enter_gc_unsafe_region_unbalanced", "ptr ptr", TRUE);
-		register_icall (mono_threads_exit_gc_unsafe_region_unbalanced, "mono_threads_exit_gc_unsafe_region_unbalanced", "void ptr ptr", TRUE);
-		register_icall (mono_threads_attach_coop, "mono_threads_attach_coop", "ptr ptr ptr", TRUE);
-		register_icall (mono_threads_detach_coop, "mono_threads_detach_coop", "void ptr ptr", TRUE);
+		register_icall (ves_icall_mono_threads_enter_gc_safe_region_unbalanced, "ves_icall_mono_threads_enter_gc_safe_region_unbalanced", "ptr ptr", TRUE);
```

Comment:
These renames create a loooooooot of line damage, let's not do it.

Reply:
ok, I thought they were wanted

Reply:
Actually I need some new name here, since I'm not keen on updating the products.

Reply:
I need some new name, so ves_icall_mono_foo, mono_foo_internal?
It is line damage but just a few lines, and I 'need' a new name -- the structuring was kinda messy and confusing with "dummy" and "cookie" and combining them into the same parameter.

Reply:
You can also use these names but declare the functions as static in the current file. This is the approach we have taken elsewhere.

Reply:
The functions are referenced in both marshal.c and marshal-ilgen.c, and currently implemented in a third file -- mono-threads-coop.c. "Fixing" that to make them static does not seem worthwhile.

Reply:
So all these functions can be implemented in `marshal.c` and declared in `marshal-internals.h`. There is no need to pollute `mono-threads-api.h` or any other non-marshal file with these declarations since they are only useful for marshalling.